src/EnergyPlus/SolarShading.cc

```diff
@@ -6387,8 +6381,8 @@ namespace SolarShading {
                 if (Surface(SurfNum).Class == SurfaceClass_TDD_Dome || SurfWinOriginalClass(SurfNum) == SurfaceClass_TDD_Diffuser)
                     SunlitFracWithoutReveal(TimeStep, HourOfDay, SurfNum) = SunLitFract; // Frames/dividers not allowed
 
-                WinTransBmBmSolar = 0.0;
-                WinTransBmDifSolar = 0.0;
+                WinTransBmBmSolar(SurfNum) = 0.0;
+                WinTransBmDifSolar(SurfNum) = 0.0;
```

Comment:
The full arrays for `WinTransBmBmSolar` and `WinTransBmDifSolar` are zeroed already in lines 6276:6277. Delete  one of these?